### DISCUSSION TITLE
New Face Swap Models (IP-Adapter-FaceID & InstantID)

### DISCUSSION THREAD

**Discussion by realmalikov** [lllyasviel/Fooocus#2050] [Ideas]
There are some new face swap models which are probably superior to the current method: IP-Adapter-FaceID and the even newer InstantID. Please implement one of them in the program.

Further information:
[IP-Adapter-FaceID Huggingface](https://huggingface.co/h94/IP-Adapter-FaceID)
[How to use IP-Adapter-FaceID with A1111](https://github.com/Mikubill/sd-webui-controlnet/pull/2434)
[InstantID GitHub](https://github.com/InstantID/InstantID)
[InstantID Huggingface](https://huggingface.co/InstantX/InstantID)

**Comment by kubilaykilinc** [comment]
InstantId is best. my opinion

**Reply by xhoxye** [reply]
How many GB of VRAM does it take GPU?

**Reply by YuWongXX** [reply]
17gb vram at least

**Comment by xhoxye** [comment]
The original version of their code usually requires 12G or more, or even 16G, 24G graphics cards

**Reply by YuWongXX** [reply]
Yes, even 8gb vram + 8gb shared vram is impossible to run. So it need at least 18gb vram to run it normally.

**Comment by mashb1t** [comment]
There are already enhancement issues for both new tools.

**Reply by lgs777** [reply]
in fooocus??

**Reply by mashb1t** [reply]
https://github.com/lllyasviel/Fooocus/issues/1522 and https://github.com/lllyasviel/Fooocus/issues/2025

**Reply by lgs777** [reply]
Thanks for the quick response, it would be great to have InstantID support too!

**Comment by YuWongXX** [comment]
InstantID is currently the only truly satisfying solution among all the non-celebrity face-swapping.
Reactor only works low resolution and cannot respond well to expressions and changes in lighting, while IPAdapter-FaceID is unable to achieve a high degree of similarity.
InstantID has some limitations, but the faces it generates are not only very similar but also very attractive, resulting in an ideal portrait effect.

**Comment by sitzbrau** [comment]
how to add InstantID to Fooocus?

**Reply by mashb1t** [reply]
use the fork mentioned in https://github.com/lllyasviel/Fooocus/issues/2025#issuecomment-1947322264 or copy its content to your machine.

**Reply by sitzbrau** [reply]
thank, i've followed the installations steps but the Inswapper tab doesn't appear:


some errore messages during the installation
```
ERROR: Could not find a version that satisfies the requirement onnxruntime-gpu==1.17.0 (from versions: none)
ERROR: No matching distribution found for onnxruntime-gpu==1.17.0
```

`DEPRECATION: torchsde 0.2.5 has a non-standard dependency specifier numpy>=1.19.*; python_version >= "3.7". pip 24.0 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of torchsde or contact the author to suggest that they release a version with a conforming dependency specifiers. Discussion can be found at https://github.com/pypa/pip/issues/12063`

```
(base) silvio@Silvios-iPro Fooocus-inswapper % .\venv\Scripts\activate     
zsh: command not found: .venvScriptsactivate
```

**Reply by mashb1t** [reply]
please open a discussion or issue as question in their repository as Fooocus core doesn't support InstantID (yet).

**Comment by orangpelupa** [comment]
the VRAM requirement has been worked around by [instantidcpu](https://github.com/rupeshs/instantidcpu), so it use huge system RAM
